src/network.ts

```diff
@@ -48,13 +58,65 @@ class Network extends VEventTarget {
     this.nodes.splice(index, 1);
   }
 
+  /**
+   * Prepare network for resolving.
+   */
+  preResolve(): void {
+    this.busy = false;
+    this.resolved = false;
+  }
+
   /**
    * Resolve all the nodes in the network.
    */
-  resolve(): void {
-    console.log(`--- ${this.name} ---`);
-    this.nodes.forEach((node) => node.resolve());
-    this.dispatchEvent(new VEvent('resolve'));
+  resolve(): Promise<void> {
+    const p = new Promise<void>((resolve, reject) => {
+      if (this.busy) reject();
+      this.busy = true;
+      console.log(`--- ${this.name} ---`);
+      if (!this.resolved) {
+        this.nodes.forEach((node) => node.preResolve());
+        this.nodes.forEach((node) => {
+          const promis = node.resolve();
+          if (promis instanceof Promise) {
+            promis.then(() => {
+              if (!(this.nodes.some((n) => !n.resolved))) {
+                this.resolved = true;
+                this.busy = false;
+                this.dispatchEvent(new VEvent('afterResolve'));
+                resolve();
+              }
+            });
+          }
+        });
+      } else {
+        reject();
+      }
+    });
+    return p;
+  }
+
+  /**
+   *
+   * @returns Boolean that states if at least one node has a state
+   */
+  step(): Promise<boolean> {
+    const promise = new Promise<boolean>((resolve) => {
+      const p = this.resolve();
```

Comment:
Unless you need to return this promise, no need to save it in variable. Just go on with `.then()` on the next line.